source/base/data_out_base.cc

```diff
@@ -3077,8 +3099,6 @@ namespace DataOutBase
             const unsigned int n_subdivisions = patch.n_subdivisions;
             const unsigned int n              = n_subdivisions + 1;
 
-            std::array<unsigned, dim> cell_order;
-            cell_order.fill(n_subdivisions);
```

Comment:
Why did you remove this? I'm slightly opposed to creating the `std::array` in every loop iteration.

Reply:
I removed it because in order for the call to the renamed function to succeed, I need to have a `std::array<unsigned,1>` in the `case 1:` branch (because I'm calling `function<1>` there), and similar in the other branches.

I do think that the way I chose is actually more compiler optimization-friendly. Here, it is really difficult for the compiler to see through the initialization and propagate `n_subdivisions` forward. On the other hand, creating the temp from an initializer list like I do in the new code, the compiler does copy-initialization and can forward-propagate the values to all of the places where it is used. Not that I think it matters: The expensive part of the function this is in is always going to be writing into the stream.